docs/source/docs/configuration.md

```diff
@@ -2,19 +2,592 @@
+
+The GlobalNet encodes the image and uses the bottleneck layer to output an affine
+transformation using a FCN.
+
+- `extract_levels`: list of positive ints (ie, the min value in `extract_levels` should
+  be >=0). Eg. [1, 2, 3, 5] will extract information at those levels (but not 4).
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+```
+
+### Loss - required
+
+This section defines the loss in training.
+
+The losses in DeepReg are defined depending on the type of network to be built, and can
+be split into three sections: imag and label losses (between moving and fixed tensors),
+and regularization losses (on the DDFs predicted).
+
+DeepReg uses `tf.keras.Model` `add_loss()` in the Registry method to add losses to the
+model, which provides some flexibility in configuration.
+
+Currently, DeepReg offers conditional, ddf/dvf and affine registration pre-built models.
+Traditionally, models have been configured with the following losses:
+
+- Conditional: label loss.
+- DDF/DVF: ddf loss, image loss, label loss.
+- Affine: ddf loss, image loss, label loss.
+
+The above sections are necessary to build a model correctly. Not passing all sections as
+defined above may raise errors. Currently you can call one loss per field eg. one label
+loss type, one image loss type and one ddf/dvf loss type.
+
+#### Image
+
+The image loss calculates dissimilarity between image tensors passed through the
+network.
+
+- `weight`: float type, weight of individual loss element in total loss function.
+- `name`: string type, one of "lncc", "ssd" or "gmi".
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+  loss:
+    image:
+      name: "lncc" # other options include "lncc", "ssd" and "gmi", for local normalised cross correlation,
+      weight: 0.1
+```
+
+The following are the default image losses. Additional arguments should be added at the
+same indent level:
+
+- `lncc`: Calls a local normalized cross-correlation type loss. Requires the following
+  arguments:
+
+  - `kernel_size`: int, optional, default=9. Kernel size or kernel sigma for
+    kernel_type="gaussian".
+  - `kernel_type`: str, optional, default="rectangular". One of "rectangular",
+    "triangular" or "gaussian"
+
+- `ssd`: Calls a sum of squared differences loss. No additional arguments required.
+
+- `gmi`: Calls a global mutual informatin loss. Requires the following arguments:
+  - `num_bins`: int, optional, default=23. Number of bins for intensity.
+  - `sigma_ratio`: float, optional, default=0.5. A hyper parameter for the gaussian
+    function.
```

Comment:
"A hyper parameter for the gaussian function." - > "A hyperparameter for the Gaussian kernel density estimation"